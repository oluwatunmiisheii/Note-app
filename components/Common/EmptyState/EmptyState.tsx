import colors from "@/utils/constants/colors";
import { View, Text, StyleSheet } from "react-native";

interface EmptyStateProps {
  title: string;
  description?: string;
}

export const EmptyState = ({ title, description }: EmptyStateProps) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.description}>{description}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "white",
  },
  title: {
    fontSize: 20,
    fontWeight: "500",
  },
  description: {
    fontSize: 16,
    color: colors.gray,
    marginTop: 10,
  },
});
