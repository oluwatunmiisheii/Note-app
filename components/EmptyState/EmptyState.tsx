import { View, Text, StyleSheet } from "react-native";

export const EmptyState = ({ text }: { text: string }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>{text}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    fontSize: 20,
    fontWeight: "500",
  },
});
