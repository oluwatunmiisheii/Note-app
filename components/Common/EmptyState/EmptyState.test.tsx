import renderer from "react-test-renderer";
import { EmptyState } from "./EmptyState";
import React from "react";
import { render } from "@testing-library/react-native";

const title = "Empty State Title";
const description = "Empty State Description";

describe(`<${EmptyState.name}>`, () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(
        React.createElement(EmptyState, {
          title,
        })
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders title passed as props", () => {
    const { getByText } = render(
      React.createElement(EmptyState, {
        title,
      })
    );
    expect(getByText(title)).toBeOnTheScreen();
  });

  describe("description", () => {
    test("renders description passed as props", () => {
      const { getByText } = render(
        React.createElement(EmptyState, {
          title,
          description,
        })
      );
      expect(getByText(description)).toBeOnTheScreen();
    });

    test("does not render description if not passed as props", () => {
      const { queryByText } = render(
        React.createElement(EmptyState, {
          title,
        })
      );
      expect(queryByText(description)).not.toBeOnTheScreen();
    });
  });
});
