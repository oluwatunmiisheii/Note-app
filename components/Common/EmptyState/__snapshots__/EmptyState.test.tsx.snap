// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "500",
      }
    }
  >
    Empty State Title
  </Text>
  <Text
    style={
      Object {
        "color": "#6c757d",
        "fontSize": 16,
        "marginTop": 10,
      }
    }
  />
</View>
`;
